homeassistant/components/wittiot/sensor.py

```diff
@@ -0,0 +1,465 @@
+        name="24H CO2",
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        device_class=SensorDeviceClass.CO2,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO2PM25,
+        sensor_type=MAIN_DATA,
+        name="CO2 PM2.5",
+        device_class=SensorDeviceClass.PM25,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO2RTPM25,
+        sensor_type=MAIN_DATA,
+        name="CO2 PM2.5 AQI",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO224HPM25,
+        sensor_type=MAIN_DATA,
+        name="CO2 24H PM2.5 AQI",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO2PM10,
+        sensor_type=MAIN_DATA,
+        name="CO2 PM10",
+        device_class=SensorDeviceClass.PM10,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO2RTPM10,
+        sensor_type=MAIN_DATA,
+        name="CO2 PM10 AQI",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_CO224HPM10,
+        sensor_type=MAIN_DATA,
+        name="CO2 24H PM10 AQI",
+        device_class=SensorDeviceClass.AQI,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LIGHTNING,
+        sensor_type=MAIN_DATA,
+        name="Thunder Last Distamce",
+        icon="mdi:lightning-bolt",
+        native_unit_of_measurement=UnitOfLength.MILES,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LIGHTNINGTIME,
+        sensor_type=MAIN_DATA,
+        name="Thunder Last Timestamp",
+        icon="mdi:lightning-bolt",
+    ),
+    WittiotSensorEntityDescription(
+        key=TYPE_LIGHTNINGNUM,
+        sensor_type=MAIN_DATA,
+        name="Thunder Daily Count",
+        icon="mdi:lightning-bolt",
+        state_class=SensorStateClass.TOTAL,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Wittiot sensor entities based on a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    sensors: list[MainDevWittiotSensor] = []
+    # Main Device Data
+    for desc in SENSOR_DESCRIPTIONS:
+        if desc.key in coordinator.data:
+            sensors.append(
+                MainDevWittiotSensor(
+                    coordinator,
+                    entry.data[CONF_HOST],
+                    entry.data[DEVICE_NAME],
+                    desc,
+                )
+            )
+
+    async_add_entities(sensors, True)
```

Comment:
```suggestion
    async_add_entities(sensors)
```

You use a coordiantor which already has data at the moment you create the entities so no need to update on add